app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt

```diff
@@ -32,6 +32,27 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
     private var selectedAccountUuid: String? = null
     private var selectedFolderId: Long? = null
 
+    override fun onSaveInstanceState(outState: Bundle) {
+            // Use local variable to avoid mutable property issues.
+            // putString supports null storage, but we will type check for consistency
+            var localSelectedAccountUuid = selectedAccountUuid
+            var localSelectedFolderId = selectedFolderId
+            var localSelectedFolderSummary = unreadFolder.summary.toString()
```

Comment:
I value consistency in general. But this feels like overdoing it. To deal with the mutability and nullability of `selectedFolderId` you can use a construct like this:

```kotlin
selectedFolderId?.let { selectedFolderId ->
    outState.putLong(SELECTED_FOLDER_ID_STATE, selectedFolderId)
}
```

You can also put this logic into an extension function to make it nicer to read at the call site:

```kotlin
override fun onSaveInstanceState(outState: Bundle) {
    // …
    outState.putLongIfPresent(SELECTED_FOLDER_ID_STATE, selectedFolderId)
    // …
}

// …

private fun Bundle.putLongIfPresent(key: String, value: Long?) {
    if (value != null) {
        putLong(key, value)
    }
}
```